client/src/components/RatingsAndReviews/RatingsAndReviews.jsx

```diff
@@ -1,75 +1,173 @@
-  const [reviewsToShow, setReviewsToShow] = useState([]);
-  const [reviewsToShowLength, setReviewsToShowLength] = useState(2);
+  // populate reviewsToShow after mounting:
+  componentDidMount() {
+    console.log('componentDidMount for reviews');
+  }
 
-  useEffect(() => {
-    setReviews(reviews);
-    setReviewsLength(reviews.length);
-    setReviewsToShow(reviews.slice(0, reviewsToShowLength));
-    setReviewsToShowLength(reviewsToShowLength + 2);
-  }, []);
+  // check if props change and send API request for more reviews
+  componentDidUpdate(prevProps) {
+    if (this.props.productId !== prevProps.productId) {
+      this.getReviews(this.props.productId);
+    }
+  }
 
-  let rating = calculateRating(reviewsMeta.ratings);
+  getReviews(id) {
+    // axios call to get reviews
+    let page = 1;
+    let count = 200;
+    let sort = 'relevance';
+    // console.log('product id in get reviews:', id);
+    axios.get(`/reviews?page=${page}&count=${count}&sort=${sort}&product_id=${id}`)
+      .then(response => {
+        this.setReviews(response.data.results);
+      })
+      .catch(err => {
+        console.log('error in Reviews request ', err);
+      });
+  }
 
-  return (
-    <div id="ratings-main" data-testid="ratings-main">
-      <div id="ratings-left-pane">
-        <div id="left-pane-title">
-          RatingsAndReviews!
-        </div>
-        <div className="rating-left" data-testid="rating-left">
-          <div className="number-rating">{rating.toFixed(1)}</div>
-          <StarReview rating={rating}/>
-        </div>
-        <div className="recommend-percent">100% of reviews recommend this product</div>
-        <div className="star-chart">
-          <img src='img/star_breakdown_placeholder.png' /></div>
-        <div className="review-size">
-          <img src='img/size_placeholder.png' /></div>
-        <div className="review-comfort">
-          <img src='img/comfort_placeholder.png' /></div>
-      </div>
-      <div id="ratings-right-pane">
-        <div id="ratings-title">{reviews.length} reviews, sorted by
-          <select name="sort-parameter" id="sort-parameter" onChange={(i) => {
-            alert('This feature only available to premium subscribers!');
-          }}>
-            <option value="relevance">relevance</option>
-            <option value="helpful">helpful</option>
-            <option value="newest">newest</option>
-          </select>
+  // reportReview(review_id) {
+
+  // }
+
+  // reviewHelpful(review_id) {
+
+
+  // }
+
+  // sorting and filter handlers:
+  // populate state with reviews:
+  setReviews(reviews) {
+    console.log('setReviews:', reviews);
+    let newReviewsToShow = reviews.slice(0, 2);
+    this.setState({
+      _reviews: reviews,
+      reviewsLength: reviews.length,
+      reviewsToShow: newReviewsToShow,
+      reviewsToShowLength: 2
+    });
+  }
+
+  setReviewsLength() {
+    let currentLength = this.state.reviews.length;
```

Comment:
I don't see a `reviews` in state, only a `_reviews`

Reply:
Good catch.